src/backend/utils/init/miscinit.c

```diff
@@ -1693,6 +1694,19 @@ process_shared_preload_libraries(void)
 				   false);
 
 	process_shared_preload_libraries_in_progress = false;
+
+	{
+		struct stat st;
+
+		if (file_read_buffer_modify_hook == NULL)
+		{
+			int file_exists = (stat("data_encryption.key", &st) ? (errno == ENOENT) : !S_ISDIR(st.st_mode));
```

Comment:
Shall we use an absolute path? e.g.,

```
snprintf(key_path, sizeof(key_path), "%s/data_encryption.key", DataDir);
```

BTW, the logic here doesn't seem quite right.

if `stat()` returns `-1` then `file_exists` will be assigned with `errno == ENOENT`. When `errno` is not `ENOENT` (which could be `EACCES`, `EBADF`, ...), the value of `file_exists` is `false`, and then the error will not be reported.